workspace.spec

```diff
@@ -572,6 +572,378 @@ module Workspace {
+
+		CRediT contributor role taxonomy: https://credit.niso.org
+
+		string contributor_type - must be either 'person' or 'organization'
+		string contributor_id (optional) - persistent unique identifier for the contributor;
+			e.g. an ORCID for an individual, or a ROR ID for an organization.
+		string given_name (optional) - first name, for individuals whose names can be split
+			into given and surnames
+		string family_name (optional) - family name, for individuals whose names can be
+			split into given and surnames.
+		string name (optional) - contributor name, in cases where the contributor is an
+			organization or a person whose name cannot easily fit into the given/family
+			name structure.
+		list<Organization> affiliations (optional) - list of organizations with which the
+			contributor is affiliated
+		list<string> contributor_roles (optional) - list of roles played by the contributor
+			when working on the resource.
+			Valid contributor_role values:
+			"Datacite:ContactPerson"
+			"Datacite:DataCollector"
+			"Datacite:DataCurator"
+			"Datacite:DataManager"
+			"Datacite:Distributor"
+			"Datacite:Editor"
+			"Datacite:HostingInstitution"
+			"Datacite:Producer"
+			"Datacite:ProjectLeader"
+			"Datacite:ProjectManager"
+			"Datacite:ProjectMember"
+			"Datacite:RegistrationAgency"
+			"Datacite:RegistrationAuthority"
+			"Datacite:RelatedPerson"
+			"Datacite:Researcher"
+			"Datacite:ResearchGroup"
+			"Datacite:RightsHolder"
+			"Datacite:Sponsor"
+			"Datacite:Supervisor"
+			"Datacite:WorkPackageLeader"
+			"Datacite:Other"
+			"CRediT:conceptualization"
+			"CRediT:data-curation"
+			"CRediT:formal-analysis"
+			"CRediT:funding-acquisition"
+			"CRediT:investigation"
+			"CRediT:methodology"
+			"CRediT:project-administration"
+			"CRediT:resources"
+			"CRediT:software"
+			"CRediT:supervision"
+			"CRediT:validation"
+			"CRediT:visualization"
+			"CRediT:writing-original-draft"
+			"CRediT:writing-review-editing"
+	*/
+	typedef structure {
+		string contributor_type;
+		string name;
+		string given_name;
+		string family_name;
+		string contributor_id;
+		list<Organization> affiliations;
+		list<string> contributor_roles;
+	} Contributor;
+
+	/*
+		Represents the credit metadata associated with a workspace object.
+
+		Required fields are:
+		- identifier
+		- resource_type
+		- titles (one or more required)
+		- contributors (one or more required)
+		- versioning information: if the resource does not have an explicit version number, one or more dates should be supplied, ideally the date of resource publication and the last update (if applicable).
+
+		string identifier - persistent unique identifier for the resource.
+			Should be in the format
+			<database name>:<identifier within database>
+			e.g.	DOI:10.46936/10.25585/60000745
+				CrossRef:IsManifestationOf
+				GO:0005456
+				HGNC:7470
+		string license (optional) - usage license for the resource.
```

Comment:
Is this the full license text or just the name of the license? For the latter, is there a CV or something like that?

Reply:
@elishawc what do you think on this issue? For at least one data source (JGI), there's a custom license that basically says that you should cite the data source if you're going to use it. Even though that is probably covered by an existing license (e.g. one of the Creative Commons licenses), it's not expressed as such on the JGI website so we should stay faithful to the source.

Reply:
Maybe a URL to a license?

Reply:
super valid Q, and I should probably check in with Grace and KJ on plans for that, but we could just link to their policy page for now. That was what Grace asked when we updated our T&Cs. 
https://jgi.doe.gov/user-programs/pmo-overview/policies/

Reply:
For now, this field can be a string or an URL. If anything changes prior to implementing the class, the spec can be updated accordingly.

Reply:
So if it looks like a url it'll be interpreted as a url IOW?